examples/apps/colorapp/servicemesh/deploy.sh

```diff
@@ -152,11 +152,45 @@ save_routes() {
     done
 }
 
+create_virtual_service() {
+    cli_input=$1
+    cmd=( $appmesh_cmd create-virtual-service \
+              ${PROFILE_OPT} \
+              --mesh-name "${MESH_NAME}" \
+              --cli-input-json "${cli_input}" \
+              --query virtualService.metadata.uid --output text )
+    print "${cmd[@]}"
+    uid=$("${cmd[@]}") || err "Unable to save virtual service" "$?"
+    print "--> ${uid}"
+}
+
+update_virtual_service() {
+    cli_input=$1
+    cmd=( $appmesh_cmd update-virtual-service \
+              ${PROFILE_OPT} \
+              --mesh-name "${MESH_NAME}" \
+              --cli-input-json "${cli_input}" \
+              --query virtualService.metadata.uid --output text )
+    print "${cmd[@]}"
+    uid=$("${cmd[@]}") || create_virtual_service "${cli_input}"
+    print "--> ${uid}"
+}
+
+save_virtual_services() {
+    print "Creating virtual services"
+    print "========================"
+    for service in $(ls ${DIR}/config/virtualservices); do
+      cli_input=$(cat ${DIR}/config/virtualservices/${service} | sed "s/@@SERVICES_DOMAIN@@/.${SERVICES_DOMAIN}/g")
+      update_virtual_service "${cli_input}"
+    done
+}
+
 main() {
     sanity_check
     save_virtual_nodes
```

Comment:
Will save_virtual_nodes here create virtual_services automatically? IMO the following order of creates makes sense to me

virtual-router -> virtual-service -> virtual-nodes -> routes

Reply:
We will create an implicit virtual service behind the scenes, but the CreateVirtualService call will flip it in active.

The reason for not putting it before VirtualNode is because of the TCP echo example. In general, it seems to make sense to follow the order of:

VirtualNodes (bottom of the tree) -> VirtualRouters -> Routes (which point to virtual nodes) -> VirtualServices (which point to VirtualRouters or VirtualNodes)